tests/testthat/test-json_reader.R

```diff
@@ -0,0 +1,36 @@
+test_path <- "custom_data/mtcars.json"
```

Comment:
Maybe rather than saving the json as this is using a built in dataset we could on the fly translate to json and save in a temp directory when testing? No feeling either way just thinking about package size (mtcars is tiny so very little impact obviously).

Reply:
One reason not to do that is that it means that we would end up testing the same version of `jsonlite` against itself. As it is, the saved `mtcars.json` file is an output of `jsonlite` but we can inspect it and read it by eye and then when we run the test (later) we are comparing use of `read_json` against a fixed input, rather than introducing the possibility that something changes in how JSON files are created (which is not something we need to test, since we don’t write JSON files anywhere).

Reply:
ah that makes sense, good point.